crates/duckdb/src/vscalar/mod.rs

```diff
@@ -132,15 +132,32 @@ where
 }
 
 impl Connection {
-    /// Register the given ScalarFunction with the current db
+    /// Register the given ScalarFunction with default state
     #[inline]
     pub fn register_scalar_function<S: VScalar>(&self, name: &str) -> crate::Result<()> {
         let set = ScalarFunctionSet::new(name);
         for signature in S::signatures() {
             let scalar_function = ScalarFunction::new(name)?;
             signature.register_with_scalar(&scalar_function);
             scalar_function.set_function(Some(scalar_func::<S>));
-            scalar_function.set_extra_info::<S::State>();
+            scalar_function.set_extra_info(S::State::default());
+            set.add_function(scalar_function)?;
+        }
+        self.db.borrow_mut().register_scalar_function_set(set)
+    }
+
+    /// Register the given ScalarFunction with custom state
+    #[inline]
+    pub fn register_scalar_function_with_state<S: VScalar>(&self, name: &str, state: S::State) -> crate::Result<()>
```

Comment:
I might be too c++-brained, but if we're going to clone the state into the box anyway, why not pass a reference here?

Reply:
You're absolutely right! Done.